solidity/contracts/Gravity.sol

```diff
@@ -434,7 +443,8 @@ contract Gravity is ReentrancyGuard, AccessControl, Pausable, Ownable {
 			}
 
 			// Check that the transaction batch is well-formed
-			if (_amounts.length != _destinations.length || _amounts.length != _fees.length) {
+			if (_payments.amounts.length != _payments.destinations.length ||
+				_payments.amounts.length != _payments.fees.length) {
```

Comment:
There is no non-zero check on the array in this calldata structs data. I'm not sure if a relayer injecting one empty record like `amounts[100,100,100,0,100,100] destination[a,b,c,d,e,f] fees[10,10,10,10,10,10]` in the struct data could cause the transaction data not match. So I think the non-zero check is needed. Correct me if I'm wrong.